theano/tensor/basic.py

```diff
@@ -3404,6 +3404,31 @@ def make_node(self, *axis_and_tensors):
 
         """
         axis, tensors = axis_and_tensors[0], axis_and_tensors[1:]
+
+        # Fix the axis if it is negative, since it crashes with fast_run
```

Comment:
I believe we should move this to _make_node_internal. This would simplify GpuJoin code at the same time.